tests/api/test_upload.py

```diff
@@ -141,6 +176,7 @@ def test_file_replacement(self):
                 "description": "My description.",
                 "categories": ["Data"],
                 "forceReplace": False,
+                "directoryLabel": "some/other",
```

Comment:
Makes sense! So you would suggest splitting this test into two separate, where one is testing the sole replacement of a file and the other in combination with changes in metadata?

Reply:
Best would be all three, I guess: replacement of file, replacement of metadata, replacement of both.
I am not sure if "replacement of metadata" in isolation works though, see https://github.com/datalad/datalad-dataverse/issues/320#issuecomment-2248988896 where I analyzed that replacing a file with itself is not possible.

Reply:
To my knowledge, only replacing the metadata is not possible since you have to supply a file with new content, as you already highlighted. One option would be to use the `forceReplace` prop. This should override the content check, but I am not 100% sure.

Reply:
Then I guess having both – replacing the file + replacing the file and metadata – is the way to go.